docs/saving/sample/EFSaving/Concurrency/Sample.cs

```diff
@@ -0,0 +1,96 @@
+    {
+        public static void Run()
+        {
+            // Ensure database is created and has a person in it
+            using (var context = new PersonContext())
+            {
+                context.Database.EnsureDeleted();
+                context.Database.EnsureCreated();
+
+                context.People.Add(new Person { FirstName = "John", LastName = "Doe" });
+                context.SaveChanges();
+            }
+
+            using (var context = new PersonContext())
+            {
+                // Fetch a person from database and change phone number
+                var person = context.People.Single(p => p.PersonId == 1);
+                person.PhoneNumber = "555-555-5555";
+
+                // Change the persons name in the database (will cause a concurrency conflict)
+                context.Database.ExecuteSqlCommand("UPDATE dbo.People SET FirstName = 'Jane' WHERE PersonId = 1");
+
+                try
+                {
+                    // Attempt to save changes to the database
+                    context.SaveChanges();
+                }
+                catch (DbUpdateConcurrencyException ex)
+                {
+                    foreach (var entry in ex.Entries)
+                    {
+                        if (entry.Entity is Person)
+                        {
+                            // Using a NoTracking query means we get the entity but it is not tracked by the context
+                            // and will not be merged with existing entities in the context.
+                            var databaseEntity = context.People.AsNoTracking().Single(p => p.PersonId == ((Person)entry.Entity).PersonId);
+                            var databaseEntry = context.Entry(databaseEntity);
+
+                            foreach (var property in entry.Metadata.GetProperties())
+                            {
+                                var proposedValue = entry.Property(property.Name).CurrentValue;
+                                var originalValue = entry.Property(property.Name).OriginalValue;
+                                var databaseValue = databaseEntry.Property(property.Name).CurrentValue;
+
+                                // TODO: Logic to decide which value should be written to database
+                                // entry.Property(property.Name).CurrentValue = <value to be saved>;
+
+                                // Update original values to 
+                                entry.Property(property.Name).OriginalValue = databaseEntry.Property(property.Name).CurrentValue;
+                            }
+                        }
+                        else
+                        {
+                            throw new NotSupportedException("Don't know how to handle concurrency conflicts for " + entry.Metadata.Name);
+                        }
+                    }
+
+                    // Retry the save operation
+                    context.SaveChanges();
```

Comment:
This really should be part of a loop, otherwise a concurrency exception here will not be handled.